src/web/test-helpers/json-helpers.ts

```diff
@@ -0,0 +1,22 @@
+/**
+ * Store helper functions for using JSON.
+ */
+
+export const replacer = (_key: any, value: any): any => {
+  if (value instanceof Map) {
+    return {
+      dataType: 'Map',
+      value: Array.from(value.entries()),
+    };
+  }
+  return value;
+};
+
+export const reviver = (_key: any, value: any): any => {
+  if (typeof value === 'object' && value !== null) {
+    if (value.dataType === 'Map') {
+      return new Map(value.value);
+    }
+  }
```

Comment:
`value` can't be falsy! Else `value?.dataType === 'Map'` will return false and the statement wouldn't be executed. I'm a bit wary about overusing optional chaining when not necessary (e.g. to avoid errors). In my opinion, better to let errors get thrown early rather than let dirty data get passed around, might lead to some nasty bugs that are hard to debug.